diff --git a/web/common/forms.php b/web/common/forms.php
index 92e3f0a..d9facf9 100755
--- a/web/common/forms.php
+++ b/web/common/forms.php
@@ -70,12 +70,15 @@ function form_full_check() {
 function validate_input(field, output, regex){
 	val = document.getElementById(field).value;
 	if (val=="") {
-		if (document.getElementById(field).getAttribute("opt")=="y")
+		if (document.getElementById(field).getAttribute("opt")=="y") {
 			document.getElementById(output).innerHTML = '';
-		else
+			document.getElementById(field).setAttribute("valid","ok");
+			ret = 1;
+		} else {
 			document.getElementById(output).innerHTML = '<img src="../../../images/share/must-icon.png">';
-		document.getElementById(field).setAttribute("valid","ko");
-		ret =-1;
+			document.getElementById(field).setAttribute("valid","ko");
+			ret = -1;
+		}
 	} else if (regex == null || val.match(new RegExp( regex,"g")) ) {
 		document.getElementById(output).innerHTML = '<img src="../../../images/share/ok_small.png">';
 		document.getElementById(field).setAttribute("valid","ok");
@@ -123,10 +126,7 @@ function form_generate_input_text($title,$tip,$id,$opt,$val,$mlen,$re) {
 	else 
 		$value = "";
 
-	if ($re==null)
-		$validate="";
-	else
-		$validate=" opt='".$opt."' oninput='validate_input(\"".$id."\", \"".$id."_ok\",\"".$re."\")'";
+	$validate=" opt='".$opt."' oninput='validate_input(\"".$id."\", \"".$id."_ok\",".($re?"\"".$re."\"":"null".")'";
 
 	print("
 		<tr>
